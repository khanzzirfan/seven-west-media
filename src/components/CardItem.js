import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Overlay from './Overlay';

const cardItemWidth = 320;
const cardItemHeight = 178;

const CardItemView = styled.div`
    display: flex;
    background-image: url(${props => props.backgroundImage});
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center center;
    width: ${cardItemWidth}px;
    height: ${cardItemHeight}px;
    margin: 10px 0 0 10px;
    position: relative;
    float: left;

    @media (max-width: 320px) {
        width: 100%;
    }
`;

const CardItem = ({ backgroundImage, iconImage, iconAlt, ImageElement, TitleElement, title }) => (
    <CardItemView backgroundImage={backgroundImage}>
        <Overlay iconAlt={iconAlt}
            iconImage={iconImage}
            ImageElement={ImageElement}
            TitleElement={TitleElement}
            title={title} />
    </CardItemView>
)

CardItem.propTypes = {
    backgroundImage: PropTypes.node,
    iconAlt: PropTypes.string,
    iconImage: PropTypes.node,
    ImageElement: PropTypes.func,
    TitleElement: PropTypes.func,
    title: PropTypes.string,
};


export default CardItem;