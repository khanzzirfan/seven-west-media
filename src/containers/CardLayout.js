import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import CardItem from '../components/CardItem';
import Image from '../components/Image';
import Title from '../components/Title';

const CardView = styled.div`
    display: flex;
    flex-wrap: wrap;
    list-style: none;
    margin: 5;
    padding: 5;
    
`;

const CardLayout = ({ data }) => (
    <CardView>
        {data.map((item, index) => (
            <CardItem key={index} 
                backgroundImage={item.backgroundImage}
                iconAlt={item.iconAlt}
                iconImage={item.iconImage}
                ImageElement={Image}
                TitleElement={Title}
                title={item.title}
            />
        ))}

    </CardView>
);


CardLayout.propTypes = {
    data: PropTypes.array,
};

export default CardLayout;