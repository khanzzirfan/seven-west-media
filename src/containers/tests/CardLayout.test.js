import React from 'react';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';

import CardLayout from '../CardLayout';
import CardItem from '../../components/CardItem';
import backgroundImage from '../../assets/tile.jpg';
import logoImage from '../../assets/logo.png';

describe('components -> CardLayout component', () => {
    let instance;
    let testRenderer;

    beforeAll(() => {
        testRenderer = TestRenderer.create(
            <CardLayout data={stubData} />);
        instance = testRenderer.root;
    });

    it('should renders correctly and match to snapshot', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it('should renders 2 card items part of card layout', () => {
        const cardItemElements = instance.findAllByType(CardItem);
        expect(cardItemElements.length).toBe(2);
    });

    it('should pass array data as props card layout', () => {
        const data = instance.props.data;
        expect(data.length).toBe(2);
        expect(data).toEqual(stubData);
    });
});


//sample data;
const stubData = [
    {
        iconAlt: 'Seven West Media',
        iconImage: logoImage,
        backgroundImage: backgroundImage,
        title: 'Home and Away',
    },
    {
        iconAlt: 'Seven West Media',
        iconImage: logoImage,
        backgroundImage: backgroundImage,
        title: 'Home and Away',
    },
];