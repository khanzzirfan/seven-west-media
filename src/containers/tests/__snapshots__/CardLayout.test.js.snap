// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components -> CardLayout component should renders correctly and match to snapshot 1`] = `
.c3 {
  margin-top: 9px;
  margin-left: 9px;
  width: 38px;
  height: 38px;
}

.c4 {
  color: white;
  margin: 0 0 0 10px;
  font-size: 20px;
  font-family: Roboto,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  background: rgba(0,0,0,0.6);
  width: 100%;
  height: 51px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-image: url(tile.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
  width: 320px;
  height: 178px;
  margin: 10px 0 0 10px;
  position: relative;
  float: left;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
  margin: 5;
  padding: 5;
}

@media (max-width:320px) {
  .c1 {
    width: 100%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="Seven West Media"
        className="c3"
        src="logo.png"
      />
      <h2
        className="c4"
      >
        Home and Away
      </h2>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="Seven West Media"
        className="c3"
        src="logo.png"
      />
      <h2
        className="c4"
      >
        Home and Away
      </h2>
    </div>
  </div>
</div>
`;
